### PR TITLE
use secure URLs (mostly homepages)

### PR DESCRIPTION
**Pull request by vszakats**
##### Instructions
- Look for and complete the section relevant to your submission. Delete the others, including these `Instructions`.
- `{{cask_file}}` represents the cask file you’re submitting/editing (if applicable).
- If there’s a checkbox you can’t complete for any reason, that’s OK. Just explain in detail why you weren’t able to do so.
#### Editing an existing cask
- [ ] Commit message includes cask’s name (and new version, if applicable).
- [ ] `brew cask audit --download {{cask_file}}` is error-free.
- [ ] `brew cask style --fix {{cask_file}}` left no offenses.

---

Let the script try to upgrade _homepage_ URLs to secure ones that may be redirected to insecure.

### CODE REVIEW

File: Casks/synergy.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
   url "http://symless.com/files/packages/synergy-v#{version.before_comma}-stable-#{version.after_comma}-MacOSX1011-x86_64.dmg"
```
Comment (reitermarkus) [inline Casks/synergy.rb]: This should be HTTPS, too.

Comment (vszakats) [inline Casks/synergy.rb]: This revealed a regression in my script. Correcting it, and there'll possibly be more catches after a re-run.

**Comment by tsparber** [comment]
Did you also check, that no redirection to an `HTTP` only url is performed? There was once an requirement, that the Cask should use `HTTPS` (for `appcast`, `url`, `homepage`) only if no intermediate server uses `HTTP`.

Btw there is also another script to "switch to https" ([`cask-switch-https`](https://github.com/caskroom/homebrew-cask/blob/master/developer/bin/cask-switch-https)).

**Comment by vszakats** [comment]
@tsparber Yes, HTTP redirection is only currently allowed for `homepage`s and URLs in comment, but not for `url`s or `appcast`s, where the redirection would be hidden from users' eyes.

But, I'd argue the current system is foolproof. Redirectors may chose either an HTTPS or HTTP redirection based on their own decision, typical scenario is when multiple mirrors are used, some of the HTTPS, some of the not. There is no sure way to detect such logic. For this reason Homebrew core implemented a different solution whereas HTTPS is _always_ used in the formulae where possible, and there is a warning shown at _download time_, telling the user that an insecure redirect has been made. Also such redirects can be disabled altogether on the user's end.

I've linked to the relevant (fairly shot) commit yesterday under another, similar commit.

**Comment by vszakats** [comment]
Link the above referred comment/commit: https://github.com/caskroom/homebrew-cask/pull/23433#issuecomment-237601601

**Comment by tsparber** [comment]
Thanks for point this out, I also think the solution implemented by `brew` is a better solution.

**Pull request merged**